src/jumpstart/saga.ts

```diff
@@ -191,3 +203,129 @@ export function* dispatchPendingERC721Transactions(
+    if (!jumpstartContractAddress) {
+      throw new Error(
+        `Jumpstart contract address for send token ${sendToken.tokenId} on network ${networkId} is not provided in dynamic config`
+      )
+    }
+
+    const createStandbyTxHandlers = []
+    const preparedTransactions = getPreparedTransactions(serializablePreparedTransactions)
+
+    // in this flow, there should only be 1 or 2 transactions. if there are 2
+    // transactions, the first one should be an approval.
+    if (preparedTransactions.length > 2) {
+      throw new Error(
+        'Received more than the maximum expected number of transactions, only 2 is allowed'
+      )
+    }
+
+    if (preparedTransactions.length === 2) {
+      const createApprovalStandbyTx = (
+        txHash: string,
+        feeCurrencyId?: string
+      ): BaseStandbyTransaction => {
+        return {
+          context: newTransactionContext(TAG, 'Approve jumpstart transaction'),
+          __typename: 'TokenApproval',
+          networkId,
+          type: TokenTransactionTypeV2.Approval,
+          transactionHash: txHash,
+          tokenId: sendToken.tokenId,
+          approvedAmount: sendAmount,
+          feeCurrencyId,
+        }
+      }
+      createStandbyTxHandlers.push(createApprovalStandbyTx)
+    }
+
+    const createStandbySendTransaction = (
+      hash: string,
+      feeCurrencyId?: string
+    ): BaseStandbyTransaction => ({
+      __typename: 'TokenTransferV3',
+      type: TokenTransactionTypeV2.Sent,
+      context: newTransactionContext(TAG, 'Send jumpstart transaction'),
+      networkId,
+      amount: {
+        value: new BigNumber(sendAmount).negated().toString(),
+        tokenAddress: sendToken.address ?? undefined,
+        tokenId: sendToken.tokenId,
+      },
+      address: jumpstartContractAddress,
+      metadata: {},
+      transactionHash: hash,
+      feeCurrencyId,
+    })
+    createStandbyTxHandlers.push(createStandbySendTransaction)
+
+    Logger.debug(
+      `${TAG}/sendJumpstartTransactionSaga`,
+      'Executing send transaction',
+      sendToken.tokenId
+    )
+
+    const txHashes = yield* call(
+      sendPreparedTransactions,
+      serializablePreparedTransactions,
+      sendToken.networkId,
+      createStandbyTxHandlers
+    )
+
+    Logger.debug(`${TAG}/sendJumpstartTransactionSaga`, 'Waiting for transaction receipts')
+    const txReceipts = yield* all(
+      txHashes.map((txHash) => {
+        return call([publicClient[networkIdToNetwork[networkId]], 'waitForTransactionReceipt'], {
+          hash: txHash,
+        })
+      })
+    )
+    txReceipts.forEach((receipt, index) => {
+      Logger.debug(
+        `${TAG}/sendJumpstartTransactionSaga`,
+        `Received transaction receipt ${index + 1} of ${txReceipts.length}`,
+        receipt
+      )
+    })
+    if (txReceipts.some((receipt) => receipt.status !== 'success')) {
+      throw new Error('One or more transactions failed')
+    }
```

Comment:
nit: we could only wait for the receipt of the TX with the highest nonce (the last one).